hm-module.nix

```diff
@@ -0,0 +1,49 @@
+self:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.programs.krewfile;
+  finalPackage = self.packages.${pkgs.system}.krewfile;
+  krewfileContent = pkgs.writeText "krewfile" (concatStringsSep "\n" cfg.plugins);
+in
+{
+  options.programs.krewfile = {
+    enable = mkEnableOption "krewfile";
+
+    installKrew = mkEnableOption "installKrew";
+
+    krewPackage = mkOption {
+      type = types.package;
+      default = pkgs.krew;
+      defaultText = literalExpression "pkgs.krew";
+      description = ''
+        Krew package to install.
+        Only relevant if `programs.krewfile.installKrew` is enabled.
+      '';
+    };
+
+    plugins = mkOption {
+      type = with types; listOf str;
+      default = [ ];
+      description = "List of plugins to be written to the krewfile.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    home.packages = ([ finalPackage ] ++ optionals cfg.installKrew [ cfg.krewPackage ]);
```

Comment:
Maybe it makes sense to only install the finalPackage if `cfg.krewPackage` is not set

Reply:
Will remove this and install krew by default